src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -79,64 +120,170 @@ bool copy_images_into_output(struct CustomNodeTensor* output, const std::vector<
+    bool convertToGrayScale = get_string_parameter("convert_to_gray_scale", params, paramsCount) == "true";
+    int targetImageHeight = get_int_parameter("target_image_height", params, paramsCount, -1);
+    int targetImageWidth = get_int_parameter("target_image_width", params, paramsCount, -1);
+    NODE_ASSERT(targetImageHeight > 0, "target image height must be larger than 0");
+    NODE_ASSERT(targetImageWidth > 0, "target image width must be larger than 0");
+
+    std::unique_lock lock(internalManagerLock);
+    CustomNodeLibraryInternalManager* internalManager = static_cast<CustomNodeLibraryInternalManager*>(*customNodeLibraryInternalManager);
+
+    // replace buffers if their sizes need to change
+    int channels = convertToGrayScale ? 1 : 3;
+    uint64_t imagesByteSize = sizeof(float) * targetImageHeight * targetImageWidth * channels * maxOutputBatch;
+    if (!(internalManager->getBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME)->getSize() == imagesByteSize &&
+            internalManager->getBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME)->getSingleBufferSize() == STREAMS_LENGTH * imagesByteSize)) {
+        internalManager->createBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME, imagesByteSize, STREAMS_LENGTH);
+    }
+
+    uint64_t coordinatesByteSize = sizeof(int32_t) * 4 * maxOutputBatch;
+    if (!(internalManager->getBuffersQueue(OUTPUT_COORDINATES_TENSOR_NAME)->getSize() == coordinatesByteSize &&
+            internalManager->getBuffersQueue(OUTPUT_COORDINATES_TENSOR_NAME)->getSingleBufferSize() == STREAMS_LENGTH * coordinatesByteSize)) {
+        internalManager->createBuffersQueue(OUTPUT_COORDINATES_TENSOR_NAME, coordinatesByteSize, STREAMS_LENGTH);
+    }
+
+    uint64_t confidenceByteSize = sizeof(float) * maxOutputBatch;
+    if (!(internalManager->getBuffersQueue(OUTPUT_CONFIDENCES_TENSOR_NAME)->getSize() == confidenceByteSize &&
+            internalManager->getBuffersQueue(OUTPUT_CONFIDENCES_TENSOR_NAME)->getSingleBufferSize() == STREAMS_LENGTH * confidenceByteSize)) {
+        internalManager->createBuffersQueue(OUTPUT_CONFIDENCES_TENSOR_NAME, confidenceByteSize, STREAMS_LENGTH);
+    }
+    return 0;
 }
 
 int initialize(void** customNodeLibraryInternalManager, const struct CustomNodeParam* params, int paramsCount) {
+    auto status = 0;
+    if (*customNodeLibraryInternalManager == nullptr) {
+        status = initializeInternalManager(customNodeLibraryInternalManager, params, paramsCount);
+    } else {
+        status = checkInternalManager(customNodeLibraryInternalManager, params, paramsCount);
+    }
+    if (status != 0)
+        return 1;
     return 0;
 }
 
 int deinitialize(void* customNodeLibraryInternalManager) {
+    // deallocate InternalManager and its contents
+    if (customNodeLibraryInternalManager != nullptr) {
+        CustomNodeLibraryInternalManager* internalManager = static_cast<CustomNodeLibraryInternalManager*>(customNodeLibraryInternalManager);
+        internalManager->clear();
```

Comment:
If we had `map<string, unique_ptr<BuffersQueue>>` we would not need the clear method

Reply:
Let me know if I get this right. If we call `delete internalManager` and in internalManager we have `std::map<str, unique_ptr>` member than on internalManager deletion every unique_ptr in map destructs object that belong to it?